Library/Homebrew/cask/lib/hbc/cli/internal_stanza.rb

```diff
@@ -68,26 +69,15 @@ def run
             next
           end
 
-          if artifact_name && !value.key?(artifact_name)
-            opoo "no such stanza '#{artifact_name}' on Cask '#{cask}'" unless quiet?
-            puts ""
-            next
-          end
-
           if stanza == :artifacts
-            value = Hash[
-              value.map do |k, v|
-                v = v.map do |a|
-                  next a.to_a if a.respond_to?(:to_a)
-                  next a.to_h if a.respond_to?(:to_h)
-                  a
-                end
-
-                [k, v]
-              end
-            ]
+            value = Hash[value.map { |v| [v.class.dsl_key, v.to_s] }]
+            value = value[artifact_name] if artifact_name
+          end
 
-            value = value.fetch(artifact_name) if artifact_name
+          if value.nil? || (value.respond_to?(:to_a) && value.to_a.empty?) ||
+             (value.respond_to?(:to_s) && value.to_s == "{}")
```

Comment:
Remove this check.

Reply:
@reitermarkus The whole check? If I remove the whole check 3 tests will break. If I remove the "{}" part of the check, the check regarding "depends_on" will break.
Please clarify your thoughts about this.

Reply:
No, just this line with the string comparison.

Reply:
OK, but how should I detect absence of the "depends_on" stanza instead in order to not break the corresponding test?

Reply:
When do you expect the string to be `"{}"`? A `Hash` responds to `to_a`, so it will be handled by the previous test.

Reply:
I expect this when querying for the "depends_on" stanza. This returns an Hbc::DSL::DependsOn object. As far as I can see this object does not respond to "to_a". It only responds to "to_s" with a string of "{}" when there is no "depends_on" stanza in the cask.

Reply:
Once https://github.com/Homebrew/brew/pull/3432 is merged, `DependsOn` will respond to `to_a`.

Reply:
Oh! Magical interventions ... ;-)